### DISCUSSION TITLE
Documenting Laravel Passport Oauth endpoints

### DISCUSSION THREAD

**Discussion by burdziakm** [dedoc/scramble#163] [Q&A]
Hello. 
Got Laravel project with Passport Oauth authentication. 
I have managed to add "oauth/token" API path along with "api" path by adding code in AppServiceProvider class boot() method:

```
public function boot(): void
    {
        Scramble::routes(function (Route $route) {
            return Str::startsWith($route->uri, ['oauth/token', 'api/']);
        });
    }
```

Scramble detected all Passport endpoints at "oauth/token" path, but it can not resolve its Request and Response. 

<img width="1680" alt="Zrzut ekranu 2023-08-2 o 12 53 06" src="https://github.com/dedoc/scramble/assets/103778082/5cb59569-cc3a-4c83-8e0a-9ee2eeb53930">

Is there a way to manually add Request and Response data to this endpoint, without publishing and modifying Passport vendor files?

Or maybe there is a way to completely manually describe such endpoint and add its informations to UI view?

**Comment by romalytvynenko** [comment]
@burdziakm Hey hey!

Great question! Let me think about the best possible option to deal with this case.

I will get back to you soon

**Reply by burdziakm** [reply]
Do You have any ideas ? 🙏

**Reply by romalytvynenko** [reply]
Hey @burdziakm 

I was exploring bunch of them, and concluded that probably the simplest way is to add `OpenAPI operations extension` in which you'd check if route is Passport's one, and add info there. Essentially hijacking default Scramble's logic and providing better data for Passport's routes.

This would be the simplest way to do that.

While there are some improvements that can be made on Scramble's side so it better understand's Passport's codebase, I feel like documenting few endpoints manually in this way will be much quicker to achieve.

Let me know if you have any questions!

**Reply by romalytvynenko** [reply]
@burdziakm https://scramble.dedoc.co/developers/extensions#openapi-operations-extension

Forgot to attach this link

**Comment by agnonym** [comment]
Hi @burdziakm, also have a Laravel project that uses Passport OAuth authentication.

If you're still interested, here's how I implemented Request body and Responses handling through an `OperationExtension` for `oauth/token` route: https://gist.github.com/tech-codivores/ddce2b10a64f06fe0b1bcd4868c17039
